packages/server/src/core/internals/getParseFn.ts

```diff
@@ -17,10 +18,15 @@ export function getParseFn<TType>(procedureParser: Parser): ParseFn<TType> {
 
   if (typeof parser.parse === 'function') {
     // ParserZodEsque
-    // ParserValibotEsque
+    // ParserValibotEsque (<= v0.12.X)
     return parser.parse.bind(parser);
   }
 
+  if (typeof parser._parse === 'function') {
```

Comment:
Wait what is the change you're making? Why can't we just call this _parse method?

Reply:
We can call `._parse`. However, `._parse` no longer throws a `ValiError` but returns an object containing either the data under `output` or all the problems found under `issues`.

https://github.com/fabian-hiller/valibot/blob/main/library/src/methods/parse/parse.ts

Reply:
We have revised the internal error handling and thus massively improved the performance.

Reply:
I could, of course, throw a standard `Error` instead of a `ValiError`. But then, all validation information would be lost, which is a problem when you want to find out why the validation failed. I can't throw a `ValiError` because I would have to import it.

Reply:
hmm okay so I think either you need to add a method to the validator that throws, or we'll have to accept a worse DX for users by letting them call `parseAsync` in a custom validator function as i show yesterday `.input(raw => parseAsync(MY_VALIBOT_VALIDATOR, raw)`

Reply:
Alternatively I could try to import `ValiError` dynamically. If someone uses Valibot, it must be installed. However, I have no experience with dynaic imports and therefore do not know how they behave in the build step.

Reply:
I have updated the code. It seems to work. At least no error occurs during the tests. I tested the following with the current version of Valibot to verify:

```ts
  if (typeof parser.parse === 'function') {
    // ParserValibotEsque (<= v0.12.X)
    if (typeof parser.schema === 'string') {
      console.log('VALIBOT 🔥🔥🔥🔥🔥🔥🔥');
      return async (value) => {
        const { parseAsync } = await import('valibot');
        return parseAsync(parser, value);
      };
    }
    // ParserZodEsque
    return parser.parse.bind(parser);
  }
```

Reply:
![screenshot](https://github.com/trpc/trpc/assets/35291865/9fa7fe61-8854-472c-961d-cc7a0f1258f0)

Reply:
The only problem i can see is a separate treatment of valibot and any other library that (in the future) puts a `_parse` on their objects will not work since that import wont work

Reply:
I understand. However, this generally applies to the current implementation of tRPC. For example, this argument could also be made with the current implementation of `ParserScaleEsque`. If tRPC wants to avoid this in the long run, individual adapters, a standardization, or an additional layer like [`TypeSchema`](https://github.com/decs/typeschema) are required.

For a quick solution, I suggest moving the code for Valibot >= v0.13.X to the end. If it should collide with another library in the near future, more conditions can be added as each schema from different libraries has different properties.

Reply:
True. What do you say @KATT & @sachinraja, is the dynamic import okay?

Reply:
No, we I don't think we should have dynamic imports, not sure if that will even work everywhere -- we need a parse function

Reply:
According to my research, `import()` is a standard supported by any environment, including Node.js and Deno. However, as described, I have no practical experience with it yet.